components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -877,6 +878,166 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+     * @throws CharonException     Exception occurred in charon level.
+     * @throws BadRequestException Exception occurred due to a bad request.
+     */
+    public User updateUser(User user, Map<String, Boolean> requiredAttributes,
+                           List<String> allSimpleMultiValuedAttributes) throws CharonException, BadRequestException {
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Updating user: " + user.getUserName());
+            }
+
+             /* Set thread local property to signal the downstream SCIMUserOperationListener
+                about the provisioning route. */
+            SCIMCommonUtils.setThreadLocalIsManagedThroughSCIMEP(true);
+            // Get user claim values.
+            Map<String, String> claims = AttributeMapper.getClaimsMap(user);
+            // Get claims mapped with simple multi-valued attributes.
+            Map<String, String> allSimpleMultiValuedClaims = new HashMap<>();
+
+            for (String simpleMultiValuedAttribute : allSimpleMultiValuedAttributes) {
+                allSimpleMultiValuedClaims.put(simpleMultiValuedAttribute, "");
+            }
+            Map<String, String> allSimpleMultiValuedClaimsList =
+                    SCIMCommonUtils.convertSCIMtoLocalDialect(allSimpleMultiValuedClaims);
+
+            // Check if username of the updating user existing in the userstore.
+            try {
+                String userStoreDomainFromSP = getUserStoreDomainFromSP();
+                SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+                User oldUser = this.getUser(user.getId(), ResourceManagerUtil.getAllAttributeURIs(schema));
+                if (userStoreDomainFromSP != null && !userStoreDomainFromSP
+                        .equalsIgnoreCase(IdentityUtil.extractDomainFromName(oldUser.getUserName()))) {
+                    String errorMessage =
+                            String.format("User :  %s is not belong to userstore %s. Hence user updating fail",
+                                    oldUser.getUserName(), userStoreDomainFromSP);
+                    throw new CharonException(errorMessage);
+                }
+                if (getUserStoreDomainFromSP() != null &&
+                        !UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME.equalsIgnoreCase(getUserStoreDomainFromSP())) {
+                    user.setUserName(IdentityUtil
+                            .addDomainToName(UserCoreUtil.removeDomainFromName(user.getUserName()),
+                                    getUserStoreDomainFromSP()));
+                }
+                // This is handled here as the IS is still not capable of updating the username via SCIM.
+                if (!StringUtils.equals(user.getUserName(), oldUser.getUserName())) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Failing the request as attempting to modify username. Old username: "
```

Comment:
`BAD REQUEST. User: <old username> attempting to modify the username as : <new username>` WDYT?

Reply:
There may be situations that come to this point other than the own person tries to change hi/her username. eg: If an admin tries to modify the username of another person, this " `User: <old username> attempting`" is misleading.
WDYT about this` BAD REQUEST. Attempting to modify username: <old username> as <new username>.`